File: go/src/Controllers/UserController.go
package Controllers

import "project/src/Core"

type UserController struct{}

func (u *UserController) Index() interface{} {
	Core.Info("User list requested")

	users := []map[string]interface{}{
		{"id": 1, "name": "John"},
		{"id": 2, "name": "Jane"},
		{"id": 3, "name": "Mike"},
	}

	return map[string]interface{}{
		"status": "success",
		"data":   users,
	}
}

File: dart/lib/controllers/user_controller.dart
import '../core/logger.dart';

class UserController {
  Map<String, dynamic> index() {
    Logger.info("User list requested");

    final users = [
      {"id": 1, "name": "John"},
      {"id": 2, "name": "Jane"},
      {"id": 3, "name": "Mike"},
    ];

    return {
      "status": "success",
      "data": users
    };
  }
}